### PR TITLE
docs (troubleshooting): reference rate limiting

### LINKED ISSUE #1101: Allow batching subscription deletions
Hey, I am trying to unsubscribe from multiple topics:

```JavaScript
    subscription.forEach(function (sub) {
      sub.delete(function (error) {
        //code
      });
    });
```

Everytime I try this, i get an unhandled error:
```
ApiError: The service was unable to fulfill your request. Please try again. [code=8a75]
at new util.ApiError (/Users/X/Y/node_modules/gcloud/lib/common/util.js:92:10)
    at Object.parseHttpRespBody (/Users/X/Y/node_modules/gcloud/lib/common/util.js:170:30)
    at Object.handleResp (/Users/X/Y/node_modules/gcloud/lib/common/util.js:110:18)
    at /Users/X/Y/node_modules/gcloud/lib/common/util.js:422:12
    at Request.onResponse [as _callback] (/Users/X/Y/node_modules/retry-request/index.js:106:7)
    at Request.self.callback (/Users/X/Y/node_modules/gcloud/node_modules/request/request.js:198:22)
    at emitTwo (events.js:87:13)
    at Request.emit (events.js:172:7)
    at Request.<anonymous> (/Users/X/Y/node_modules/gcloud/node_modules/request/request.js:1035:10)
    at emitOne (events.js:82:20)
```

Any idea?

EDIT: interesting enough, even though it throws an error, it still seems to delete the subscriptions.

### PR DESCRIPTION
**Pull request by stephenplusplus**
Fixes #1101

### CODE REVIEW

File: docs/troubleshooting.md [hunk 1]
```diff
@@ -5,3 +5,44 @@ There are many types of keys that can be created through the Developer's Console
...
+
+## My requests are returning errors instructing me to retry the request.
+
+If you ever see an error like this, you might be pinging the API too rapidly, hitting a rate limit:
+
+```sh
+ApiError: The service was unable to fulfill your request. Please try again.
+```
+
+Each API call might have its own requirements for what constitutes reaching the rate limit. We solve this across most of the requests we make by using an exponential backoff retry stragety. However, if you make several requests at once, you will inevitably reach that limit.
```
Comment (callmehiphop) [inline docs/troubleshooting.md]: strategy*

File: docs/troubleshooting.md [hunk 2]
```diff
@@ -5,3 +5,44 @@ There are many types of keys that can be created through the Developer's Console
...
+
+As an example, if you need to delete multiple Pub/Sub subscriptions at a time, you would likely do something like:
```
Comment (callmehiphop) [inline docs/troubleshooting.md]: *"likely do something like"* - reads kind of funny to me.

**Comment by callmehiphop** [comment]
Is this a good time to bring up testing markdown js snippets?

**Comment by stephenplusplus** [comment]
Haha, it probably deserves its own issue. But that sounds like a good idea!

**Comment by stephenplusplus** [comment]
Fixed! Would it be crazy to spell check our docs, too?

**Comment by callmehiphop** [comment]
I don't think so! We could always extend the lint task with something like [write-good](https://github.com/btford/write-good)?

**Comment by stephenplusplus** [comment]
That looks pretty slick. Though it doesn't appear to spell check. I'll open an issue so we can talk about it more.

**Pull request merged**